### PR TITLE
cmake: Only go into grc/ subdirs when ENABLE_GRC=ON

### LINKED ISSUE #2154: CMake enables gr-uhd but compilation fails due to missing Python module mako
On a system without `python-mako` CMake enables `gr-uhd`, however compilation fails due to missing Python module `mako`. I can reproduce this error on Ubuntu 18.04 LTS and Arch Linux.

### PR DESCRIPTION
**Pull request by mbr0wn**
Fixes #2154.

### CODE REVIEW

File: gr-blocks/CMakeLists.txt [hunk 1]
```diff
@@ -54,11 +54,13 @@ add_subdirectory(lib)
...
 #endif(ENABLE_TESTING)
 if(ENABLE_PYTHON)
-     add_subdirectory(python/blocks)
-     add_subdirectory(swig)
-     add_subdirectory(grc)
-     add_subdirectory(docs)
-     add_subdirectory(examples)
+    add_subdirectory(python/blocks)
+    add_subdirectory(swig)
+    if(ENABLE_GRC)
```
Comment (michaelld) [inline gr-blocks/CMakeLists.txt]: I'd prefer that this ENABLE_GRC is outside the scope of ENABLE_PYTHON, to match all of the others.

File: gr-blocks/CMakeLists.txt [hunk 2]
```diff
@@ -54,11 +54,13 @@ add_subdirectory(lib)
 #  add_subdirectory(tests)
 #endif(ENABLE_TESTING)
 if(ENABLE_PYTHON)
-     add_subdirectory(python/blocks)
-     add_subdirectory(swig)
-     add_subdirectory(grc)
-     add_subdirectory(docs)
-     add_subdirectory(examples)
+    add_subdirectory(python/blocks)
```
Comment (michaelld) [inline gr-blocks/CMakeLists.txt]: I'd prefer that whitespace changes are minimally in a separate commit, if not a separate PR.

File: gr-uhd/grc/gen_uhd_usrp_blocks.py [hunk 3]
```diff
@@ -114,7 +114,7 @@
 -   domain: message
     id: command
     optional: true
-    hide: ${'$'}{hide_cmd_port}
+    hide: ${'$'}{'all' if bool(eval(hide_cmd_port)) else 'none'}
```
Comment (michaelld) [inline gr-uhd/grc/gen_uhd_usrp_blocks.py]: I'd greatly prefer that this change is in a different PR.

File: gr-utils/python/modtool/gr-newmod/CMakeLists.txt [hunk 4]
```diff
@@ -190,11 +190,15 @@ add_custom_target(uninstall
 ########################################################################
 add_subdirectory(include/howto)
 add_subdirectory(lib)
-add_subdirectory(swig)
-add_subdirectory(python)
-add_subdirectory(grc)
 add_subdirectory(apps)
 add_subdirectory(docs)
+if(ENABLE_PYTHON)
```
Comment (michaelld) [inline gr-utils/python/modtool/gr-newmod/CMakeLists.txt]: I'd prefer that this change for ENABLE_PYTHON is minimally in a separate commit, if not a separate PR.

**Comment by michaelld** [comment]
I like the concepts in this change, overall. I'd prefer some changes be in different commits and/or a different PR entirely ... but, if not then I'm OK with going with this as-is.

File: gr-uhd/grc/gen_uhd_usrp_blocks.py (hunk 3 above)
Comment (mbr0wn) [reply gr-uhd/grc/gen_uhd_usrp_blocks.py]: This was a mistake. Good catch.

File: gr-blocks/CMakeLists.txt (hunk 2 above)
Comment (mbr0wn) [reply gr-blocks/CMakeLists.txt]: I can undo that. First, I had the ENABLE_GRC inside the ENABLE_PYTHON, where it was hard to separate the two.

**Comment by mbr0wn** [comment]
@michaelld Thanks for the comments! Please use the 'Changes Requested' option from the review for this thing; it's easier to parse on the PR overview page.

**Comment by mbr0wn** [comment]
@michaelld Addressed all of your comments.

**Comment by mbr0wn** [comment]
It's not a big difference, except that it more clearly communicates change requests/approvals.
